### ISSUE TITLE
Precompiled binary request for Ubuntu 22.04

### ISSUE THREAD

**Issue by kristirme** [nutjunkie/IQmol3#14]
Is it possible to get a precompiled binary that is compatible with the recent LTS version of Ubuntu? It looks like the most recent binary for linux is IQmol version 2.14, which is not compatible with the updated operating system, and trying to compile from source code is currently giving linking errors with libm-2.35 (error below). I will evenutally figure out the linking issue, but it would be nice to just have a version of the binary that works again. Thank you!

[100%] Linking CXX executable bin/IQmol
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libm-2.35.a(e_pow.o): warning: relocation against `_dl_x86_cpu_features' in read-only section `.text'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libm-2.35.a(s_atan.o): in function `__atan_ifunc':
(.text+0x526): undefined reference to `_dl_x86_cpu_features'
/usr/bin/ld: (.text+0x531): undefined reference to `_dl_x86_cpu_features'
/usr/bin/ld: (.text+0x53a): undefined reference to `_dl_x86_cpu_features'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libm-2.35.a(s_ceil.o): in function `__ceil_ifunc':
(.text+0x6): undefined reference to `_dl_x86_cpu_features'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libm-2.35.a(s_floor.o): in function `__floor_ifunc':
(.text+0x6): undefined reference to `_dl_x86_cpu_features'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libm-2.35.a(s_sin.o):(.text+0x12c6): more undefined references to `_dl_x86_cpu_features' follow
/usr/bin/ld: warning: creating DT_TEXTREL in a PIE
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/IQmol.dir/build.make:299: bin/IQmol] Error 1
make[1]: *** [CMakeFiles/Makefile2:907: CMakeFiles/IQmol.dir/all] Error 2
make: *** [Makefile:166: all] Error 2

**Comment by felixplasser** [comment]
I had the same error message on my machine. It works if I use dynamic linking for libma. I use the following process:

-  `make VERBOSE=ON`
This gives me the full linking line

- I delete `.../libma.a` from the linking line and add `-lm`
- Then I run this new command:
`/usr/bin/c++   -Wno-deprecated -O3 -DNDEBUG   CMakeFiles/IQmol.dir/IQmol_autogen/mocs_compilation.cpp.o CMakeFiles/IQmol.dir/src/main.C.o CMakeFiles/IQmol.dir/qrc_IQmol.cpp.o  -o bin/IQmol  -Wl,-rpath,/PhotoChem/programs/QChem/IQMol/GIT3/build/Build/lib src/Main/libMain.a src/Process/libProcess.a src/Parser/libParser.a src/Network/libNetwork.a src/Qui/libQui.a src/Layer/libLayer.a src/Viewer/libViewer.a src/Configurator/libConfigurator.a src/Data/libData.a src/Grid/libGrid.a src/Util/libUtil.a src/Math/libMath.a src/Plot/libPlot.a src/Fortran/libFort.a modules/openbabel/src/libopenbabel.a modules/libssh2/src/libssh2.a modules/yaml-cpp/libyaml-cpp.a modules/libQGLViewer.a /usr/lib/x86_64-linux-gnu/libQt5PrintSupport.so.5.12.8 Build/lib/libOpenMeshTools.so.8.0 /usr/lib/x86_64-linux-gnu/libboost_iostreams.a /usr/lib/x86_64-linux-gnu/libboost_serialization.a /usr/lib/x86_64-linux-gnu/libOpenGL.so /usr/lib/x86_64-linux-gnu/libGLX.so /usr/lib/x86_64-linux-gnu/libGLU.so /usr/lib/x86_64-linux-gnu/libz.so /usr/lib/x86_64-linux-gnu/libQt5Sql.so.5.12.8 src/Process/libProcess.a src/Network/libNetwork.a /usr/lib/x86_64-linux-gnu/libQt5Network.so.5.12.8 /usr/lib/x86_64-linux-gnu/libssl.so /usr/lib/x86_64-linux-gnu/libcrypto.so src/Layer/libLayer.a /usr/lib/x86_64-linux-gnu/libQt5Network.so.5.12.8 src/Parser/libParser.a /usr/lib/x86_64-linux-gnu/libQt5Network.so.5.12.8 src/Configurator/libConfigurator.a src/Plot/libPlot.a src/Data/libData.a src/Math/libMath.a modules/libQGLViewer.a src/Util/libUtil.a /usr/lib/x86_64-linux-gnu/libQt5Xml.so.5.12.8 /usr/lib/x86_64-linux-gnu/libQt5OpenGL.so.5.12.8 modules/openbabel/external/maeparser-v1.2.3/maeparser/libmaeparser.a /usr/lib/x86_64-linux-gnu/libboost_iostreams.so.1.71.0 /usr/lib/x86_64-linux-gnu/libssl.so /usr/lib/x86_64-linux-gnu/libcrypto.so /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5.12.8 /usr/lib/x86_64-linux-gnu/libQt5Gui.so.5.12.8 /usr/lib/x86_64-linux-gnu/libQt5Core.so.5.12.8 Build/lib/libOpenMeshCore.so.8.0 /usr/lib/x86_64-linux-gnu/libz.so -lgfortran -lquadmath -lm`

**Comment by felixplasser** [comment]
Then IQmol opens up alright. But sometimes it crashes with a segmentation fault when I click on atoms (both the precompiled binaries and my new installation).

The error message looks like this:
~~~~
OBMol::SetImplicitValencePerceived() deprecated
DEBUG 2023-05-09T14:41:19.190 Stoping animation 
FATAL 2023-05-09T14:41:20.564 Fatal error encountered: signal  11 
iqmol3(+0x15675c)[0x5608c09b575c]
/lib/x86_64-linux-gnu/libc.so.6(+0x43090)[0x7f053eb75090]
/usr/lib/x86_64-linux-gnu/dri/i965_dri.so(+0x1d0dfe)[0x7f052a6c7dfe]
...
/usr/lib/x86_64-linux-gnu/dri/i965_dri.so(+0x3c8f0d)[0x7f052a8bff0d]
/usr/lib/x86_64-linux-gnu/dri/i965_dri.so(+0x3c946f)[0x7f052a8c046f]
FATAL 2023-05-09T14:41:20.565 Message:  Segmentation fault 
Qt has caught an exception thrown from an event handler. Throwing
exceptions from an event handler is not supported in Qt.
You must not let any exception whatsoever propagate through Qt code.
If that is not possible, in Qt 5 you must at least reimplement
QCoreApplication::notify() and catch all exceptions there.

FATAL 2023-05-09T14:41:20.565 Signal caught 
FATAL 2023-05-09T14:41:20.565 ------- EXCEPTION ENCOUNTERED ------- 
Segmentation fault
~~~~

**Comment by kristirme** [comment]
I have gotten it to work using the suggestions by felixplasser, but it would still be nice to have a precompiled ubuntu binary available again when possible.

**Comment by tim-the-wizard** [comment]
Still no such binary? Attempting to install on Mint 22.2 (Ubuntu 24.04) fails due to dependency issues. Part of the issue seems to be dependency on qt5, which is not available on these newer distros. Trying to build IQMol3 also fails because it relies on qt5.

Attempting to install as per user guide just results in removal of package when apt cant resolve dependencies:
```sh
sudo sudo dpkg -i iqmol_3.2.2-1_amd64.deb
Selecting previously unselected package iqmol.
(Reading database ... 651627 files and directories currently installed.)
Preparing to unpack iqmol_3.2.2-1_amd64.deb ...
Unpacking iqmol (3.2.2-1) ...
dpkg: dependency problems prevent configuration of iqmol:
 iqmol depends on libyaml-cpp0.8; however:
  Package libyaml-cpp0.8 is not installed.
 iqmol depends on libqt5printsupport5t64; however:
  Package libqt5printsupport5t64 is not installed.
 iqmol depends on libqt5sql5t64; however:
  Package libqt5sql5t64 is not installed.
 iqmol depends on libqglviewer2-qt5t64; however:
  Package libqglviewer2-qt5t64 is not installed.
 iqmol depends on libopenmesh1; however:
  Package libopenmesh1 is not installed.
 iqmol depends on libopenbabel7; however:
  Package libopenbabel7 is not installed.
 iqmol depends on libopenmesh1; however:
  Package libopenmesh1 is not installed.
 iqmol depends on libc6-i386; however:
  Package libc6-i386 is not installed.
 iqmol depends on libc6-i386; however:
  Package libc6-i386 is not installed.
 iqmol depends on diversion; however:
  Package diversion is not installed.
 iqmol depends on lib32stdc++6; however:
  Package lib32stdc++6 is not installed.
 iqmol depends on lib32gcc-s1; however:
  Package lib32gcc-s1 is not installed.
 iqmol depends on libqt5xml5t64; however:
  Package libqt5xml5t64 is not installed.
 iqmol depends on libc6-i386; however:
  Package libc6-i386 is not installed.

dpkg: error processing package iqmol (--install):
 dependency problems - leaving unconfigured
Processing triggers for mailcap (3.70+nmu1ubuntu1.24.04.1) ...
Processing triggers for gnome-menus (3.36.0-1.1ubuntu3) ...
Processing triggers for mate-menus (1.26.1+mint1) ...
Processing triggers for desktop-file-utils (0.27-2build1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Errors were encountered while processing:
 iqmol

sudo apt -f install
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Correcting dependencies... Done
The following packages were automatically installed and are no longer required:
  cmake-data gfortran-13 gfortran-13-x86-64-linux-gnu gfortran-x86-64-linux-gnu libgfortran-13-dev libjsoncpp25 libqt6concurrent6t64 libqt6openglwidgets6t64 libqt6printsupport6t64
  libqt6sql6-sqlite libqt6sql6t64 libqt6test6t64 libqt6widgets6t64 libqt6xml6t64 librhash0 libvulkan-dev qmake6 qmake6-bin qt6-base-dev-tools
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
  iqmol
0 to upgrade, 0 to newly install, 1 to remove and 116 not to upgrade.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] 
```

Building from source gets stuck when it can't find the qt5 config cmake file, even with qt6 installed.
